### PR TITLE
Add new frontend

### PR DESCRIPTION
**Pull request by 0xVikas**
Adds new features: #40

### CODE REVIEW

**Comment by Kushagra-0801** [comment]
The commit message should be in imperative mood.

> ~Adds new frontend~
> Add new frontend

**Comment by Kushagra-0801** [comment]
Your commit messages do not tell anything. Either squash them into one commit or give meaningful names

**Comment by Kushagra-0801** [comment]
There is still no template inheritance

File: complaints/views.py [hunk 1]
```diff
@@ -154,7 +154,16 @@ def request_unblock(request):
                 user_email=request.user.email,
             )
             messages.success(request, "Your Request has been Successfully Registered")
-            return render(request, "registration/home.html", context={"title": "home"})
+            return render(request, "complaints/request_unblock.html", context={"title": "home"})
```
Comment (Aviral14) [inline complaints/views.py]: No need to send title from here as you have already used the title tag in the html files.

File: pages/views.py [hunk 2]
```diff
@@ -8,7 +8,7 @@ def home(request):
         return redirect("account-login")
 
     return render(
-        request, "registration/home.html", context={"title": "CCIT Complaint Portal"}
+        request, "complaints/complaints_register.html", context={"title": "CCIT Complaint Portal"}
     )
```
Comment (Aviral14) [inline pages/views.py]: Remove unnecessary context .

File: pages/templates/registration/contact.html [hunk 3]
```diff
@@ -1,4 +1,66 @@
...
+    <div class="collapse navbar-collapse" id="myNavbar">
+    <ul class="nav navbar-nav">
+	{% if user.is_authenticated %} 
+	{% if user.is_staff or user.is_nucleus %}
+	  <li><a href="/complaints/my-tickets">All Complaints</a></li>
+	{%else%}
+	  <li><a href="/complaints/my-tickets">My Complaints</a></li>
+	{%endif%}
+	{% if not user.is_staff and not user.is_nucleus %}
+      <li class=""><a href="/complaints/add-tickets">Register Complaints</a></li>
```
Comment (Aviral14) [inline pages/templates/registration/contact.html]: Use reverse lookup approach for urls

File: complaints/templates/complaints/request_unblock.html [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+               <div class="col-md-9">
+               <input id="url" name="url" type="url" placeholder="http://www.example.com" class="form-control">
+               <!-- render_field form2.url id="" class="form-control" %}-->
+               </div>
+               </div>
+               <div class="form-group">
+               <label class="col-md-3 control-label" for="message">Details</label>
+               <div class="col-md-9">
+               <textarea class="form-control" id="reason" name="reason" a="true" placeholder="Please enter your message here..." rows="5"></textarea>
+               <!-- render_field form2.reason id="" class="form-control" style="height:115px;" row="5" placeholder="Please enter your messe here..." %}-->
```
Comment (Aviral14) [inline complaints/templates/complaints/request_unblock.html]: Remove unnecessary comments

File: complaints/templates/complaints/register_complaints.html [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    
+            <!-- Name input-->
+			<div class="form-group">
+			<center>
+			<select id="option" class="form-control form-control-sm" a="true" style="width:200px;">
+                  <option value="none">Choose a Category</option>
+                  <option value="wifi">WiFi</option>
+                  <option value="lan">Lan</option>
+                  <option value="firewall">Sophos Firewall</option>
+                  <option value="url_unblock">URL Unblock req.</option>
```
Comment (Aviral14) [inline complaints/templates/complaints/register_complaints.html]: As there is another separate tab for url ublock requests,remove that from here.

File: complaints/templates/complaints/my_complaints.html [hunk 6]
```diff
@@ -0,0 +1,43 @@
+<!DOCTYPE html>
```
Comment (Aviral14) [inline complaints/templates/complaints/my_complaints.html]: What is this file for? You haven't incuded any data here.Remove thhis file if it is unnecessary.

File: CCIT_Portal/templates/account/login.html [hunk 7]
```diff
@@ -1,44 +1,113 @@
-{% extends "account/base.html" %}
-
+<!DOCTYPE html>
+{% load static %}
 {% load i18n %}
 {% load account socialaccount %}
```
Comment (Aviral14) [inline CCIT_Portal/templates/account/login.html]: Remove all the tags with i18 or trans and put there the actual things you want to display

File: complaints/templates/complaints/complaints_register.html [hunk 8]
```diff
@@ -1,11 +1,139 @@
...
-<form method="post" enctype="multipart/form-data">
-    {% csrf_token %}
-    {{ form.as_p }}
-    <button type="submit">Upload</button>
-</form>
-
-<p><a href='http://localhost:8000/home/'>Return to Home Page</a></p>
-{% endblock %}
\ No newline at end of file
+<!DOCTYPE html>
```
Comment (Aviral14) [inline complaints/templates/complaints/complaints_register.html]: Please use template inheritance

File: complaints/templates/complaints/complaints_register.html [hunk 9]
```diff
@@ -1,11 +1,139 @@
...
+                           {{ message }}
+                           {% if message|length %}<br><br>{%endif%}
+                           <select id="option" name="category" class="form-control form-control-sm" a="true" style="width:200px;">
+                              <option value="none">Choose a Category</option>
+                              <option value="wifi">WiFi</option>
+                              <option value="lan">Lan</option>
+                              <option value="firewall">Sophos Firewall</option>
+                              <!--<option value="url_unblock">URL Unblock req.</option>-->
+                           </select>
+                           <!-- render_field form.category id="option" class="form-control form-control-sm" a="true" style="width:200px;" placeholder="Choose a category" %}-->
```
Comment (Aviral14) [inline complaints/templates/complaints/complaints_register.html]: Remove unnecessary comments

File: complaints/templates/complaints/contact_before.html [hunk 10]
```diff
@@ -0,0 +1,41 @@
+<!DOCTYPE html>
```
Comment (Aviral14) [inline complaints/templates/complaints/contact_before.html]: You already have contact.html.Then what is this file for?
Please remove such duplicate and unnecessary files.

File: complaints/templates/complaints/contact.html [hunk 11]
```diff
@@ -1,35 +1,6 @@
...
-      </button>
-	  <img class="navbar-brand" src="/static/bitslogo.png"></img>
-      <a class="navbar-brand" href="#">CCIT BPHC</a>
-    </div>
-    <div class="collapse navbar-collapse" id="myNavbar">
-    <ul class="nav navbar-nav">
+{% block content %}
 	{% if user.is_authenticated %} 
 	{% if user.is_staff or user.is_nucleus %}
 	  <li><a href="/complaints/my-tickets">All Complaints</a></li>
```
Comment (Aviral14) [inline complaints/templates/complaints/contact.html]: Use Reverse lookup!

File: pages/templates/registration/contact.html [hunk 12]
```diff
@@ -1,35 +1,6 @@
...
-      </button>
-	  <img class="navbar-brand" src="/static/bitslogo.png"></img>
-      <a class="navbar-brand" href="#">CCIT BPHC</a>
-    </div>
-    <div class="collapse navbar-collapse" id="myNavbar">
-    <ul class="nav navbar-nav">
+{% block content %}
 	{% if user.is_authenticated %} 
 	{% if user.is_staff or user.is_nucleus %}
 	  <li><a href="/complaints/my-tickets">All Complaints</a></li>
```
Comment (Aviral14) [inline pages/templates/registration/contact.html]: Use Reverse Lookup!

File: CCIT_Portal/templates/account/login.html [hunk 13]
```diff
@@ -1,44 +1,113 @@
...
+        <span class="icon-bar"></span>
+        <span class="icon-bar"></span>
+        <span class="icon-bar"></span>                        
+      </button>
+	  <img class="navbar-brand" src="/static/bitslogo.png"></img>
+      <a class="navbar-brand" href="#">CCIT BPHC</a>
+    </div>
+    <div class="collapse navbar-collapse" id="myNavbar">
+    <ul class="nav navbar-nav">
+      <li><a href="/contact">Contact Us</a></li>
```
Comment (Kushagra-0801) [inline CCIT_Portal/templates/account/login.html]: add reverse url lookup

File: CCIT_Portal/templates/account/login.html [hunk 14]
```diff
@@ -1,44 +1,113 @@
...
+    </ul>
+    <ul class="nav navbar-nav navbar-right">
+      <li><a href="/login"><span class="glyphicon glyphicon-user"></span> Login</a></li>
```
Comment (Kushagra-0801) [inline CCIT_Portal/templates/account/login.html]: url here too

File: complaints/static/contact.html [hunk 15]
```diff
@@ -0,0 +1,59 @@
+<!DOCTYPE html>
```
Comment (Kushagra-0801) [inline complaints/static/contact.html]: no inheritence

File: complaints/templates/complaints/base.html [hunk 16]
```diff
@@ -0,0 +1,32 @@
...
+        <span class="icon-bar"></span>
+        <span class="icon-bar"></span>                        
+      </button>
+	  <img class="navbar-brand" src="/static/bitslogo.png"></img>
+      <a class="navbar-brand" href="#">CCIT BPHC</a>
+    </div>
+    <div class="collapse navbar-collapse" id="myNavbar">
+    <ul class="nav navbar-nav">
+	{% block content %}
+	{%endblock%}
```
Comment (Kushagra-0801) [inline complaints/templates/complaints/base.html]: tags are not balanced. Where's </body>, </html> etc?

File: complaints/templates/complaints/complaints_register.html [hunk 17]
```diff
@@ -1,11 +1,104 @@
...
+                           <label class="col-md-3 control-label" for="message" >Complaint Details</label>
+                           <div class="col-md-9">
+                              <textarea class="form-control" a="true" id="message" name="remark" placeholder="Please enter your message here..." rows="5"></textarea>
+                           </div>
+                        </div>
+                        <div class="form-group row">
+                           <label class="col-md-3 control-label" for="message">-</label>
+                           <div class="col-md-9">
+                              <label for="default" class="btn btn-default" onclick="urg_fun()" style="position:relative;">
+                                 <b>Urgency</b><input type="checkbox" id="default" class="badgebox"><!-- render_field form.urgency id="default" class="badgebox" %}--><span class="badge">&check;</span>
```
Comment (Kushagra-0801) [inline complaints/templates/complaints/complaints_register.html]: useless comment and isn't there an attribute on the checkbox input to mark / unmark it?

**Comment by Kushagra-0801** [comment]
@0xVikas fix these things. Also can we have an indication (like a star or bold) for required fields?

File: complaints/templates/complaints/base.html (hunk 16 above)
Comment (Kushagra-0801) [reply complaints/templates/complaints/base.html]: testing testing

<div style="position: absolute; margin: 50px; padding: 50px; size: 20em"> <b> Testing whether github has a bug </b> </div>

Comment (Kushagra-0801) [reply complaints/templates/complaints/base.html]: <form method="post"> 
    <input type="textbox" placeholder="bug">
<input type="submit" value="submit"> Submit</input>
</form>

File: complaints/templates/complaints/base.html [hunk 18]
```diff
@@ -27,4 +27,6 @@
       <a class="navbar-brand" href="#">CCIT BPHC</a>
     </div>
     <div class="collapse navbar-collapse" id="myNavbar">
-    <ul class="nav navbar-nav">
\ No newline at end of file
+    <ul class="nav navbar-nav">
+	{% block content %}
+	{%endblock%}
```
Comment (Aviral14) [inline complaints/templates/complaints/base.html]: %3Chtml%3E%0A%3Cbody%3E%0A%0A%3Ch2%3EHTML%20Forms%3C%2Fh2%3E%0A%0A%3Cform%20action%3D%22%2Faction_page.php%22%3E%0A%20%20First%20name%3A%3Cbr%3E%0A%20%20%3Cinput%20type%3D%22text%22%20name%3D%22firstname%22%20value%3D%22Mickey%22%3E%0A%20%20%3Cbr%3E%0A%20%20Last%20name%3A%3Cbr%3E%0A%20%20%3Cinput%20type%3D%22text%22%20name%3D%22lastname%22%20value%3D%22Mouse%22%3E%0A%20%20%3Cbr%3E%3Cbr%3E%0A%20%20%3Cinput%20type%3D%22submit%22%20value%3D%22Submit%22%3E%0A%3C%2Fform%3E%20%0A%0A%3Cp%3EIf%20you%20click%20the%20%22Submit%22%20button%2C%20the%20form-data%20will%20be%20sent%20to%20a%20page%20called%20%22%2Faction_page.php%22.%3C%2Fp%3E%0A%0A%3C%2Fbody%3E%0A%3C%2Fhtml%3E%0A

Comment (Aviral14) [inline complaints/templates/complaints/base.html]: <html>
<body>

<h2>HTML Forms</h2>

<form action="/action_page.php">
  First name:<br>
  <input type="text" name="firstname" value="Mickey">
  <br>
  Last name:<br>
  <input type="text" name="lastname" value="Mouse">
  <br><br>
  <input type="submit" value="Submit">
</form> 

<p>If you click the "Submit" button, the form-data will be sent to a page called "/action_page.php".</p>

</body>
</html>

Comment (Aviral14) [inline complaints/templates/complaints/base.html]: %3Cform%20action%3D%22%2Faction_page.php%22%3E%0A%20%20First%20name%3A%3Cbr%3E%0A%20%20%3Cinput%20type%3D%22text%22%20name%3D%22firstname%22%20value%3D%22Mickey%22%3E%0A%20%20%3Cbr%3E%0A%20%20Last%20name%3A%3Cbr%3E%0A%20%20%3Cinput%20type%3D%22text%22%20name%3D%22lastname%22%20value%3D%22Mouse%22%3E%0A%20%20%3Cbr%3E%3Cbr%3E%0A%20%20%3Cinput%20type%3D%22submit%22%20value%3D%22Submit%22%3E%0A%3C%2Fform%3E%0A

File: complaints/templates/complaints/base.html (hunk 16 above)
Comment (0xVikas) [reply complaints/templates/complaints/base.html]: Can't balance all the tags in the base.html ,they are balanced in the respective pages.

Comment (Kushagra-0801) [reply complaints/templates/complaints/base.html]: Why?

Comment (0xVikas) [reply complaints/templates/complaints/base.html]: because the code inside these tags varies in each page.

Comment (Kushagra-0801) [reply complaints/templates/complaints/base.html]: And why does that mean that you cannot close an open element?

File: complaints/static/contact.html [hunk 19]
```diff
@@ -55,5 +33,6 @@ <h4>Name : 91XXXXXXXXXX</h4>
 </div>
 </div>
 
+{% endblock %}
```
Comment (Aviral14) [inline complaints/static/contact.html]: closing body and html tags after endblock are in each file.So just put them in base.html to remove code duplicacy

Comment (Aviral14) [reply complaints/static/contact.html]: Also, Please squash all your commits in this branch into one before pushing.

File: complaints/templates/complaints/login.html [hunk 20]
```diff
@@ -0,0 +1,58 @@
...
+            <center><button type="button" class="btn" style="width:50px;display:block;"><i class="fa fa-unlock-alt" style="font-size:18px"></i></button></center>
+          </div>
+        </form>        
+       
+    </div>    
+  </div>
+</div>
+
+</body>
+</html>
```
Comment (Aviral14) [inline complaints/templates/complaints/login.html]: What is this file for? When we already have a login.html in accounts folder.Remove this if it is a duplicate file.

File: complaints/views.py [hunk 21]
```diff
@@ -75,7 +75,7 @@ def register_complaint(request):
             )
 
             messages.success(request, "Your Complaint has been Successfully Registered")
-            return render(request, "registration/home.html", context={"title": "home"})
+            return render(request, "complaints/complaints_register.html", context={"title": "home"})
```
Comment (Aviral14) [inline complaints/views.py]: Remove this unnecessary context.

File: complaints/views.py (hunk 1 above)
Comment (Aviral14) [reply complaints/views.py]: Remove this context too!

File: pages/views.py (hunk 2 above)
Comment (Aviral14) [reply pages/views.py]: This also!

**Pull request merged**